### ISSUE TITLE
 [a11y]: ProgressIndicator [default] - current step bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15264]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Current Step should be disabled since it is active

Current step is not disabled though it is active status which misleads the SR users

### Recommended Fix
Current Step needs to be disabled

If the screen reader does not announce the status change in a page element, it’s difficult for users to understand it on their own. This defect mainly impacts users who are blind. 

### WCAG 2.1 Violation
4.1.2 Name, Role, Value

**Comment by 2nikhiltom** [comment]
The issue was retested  (in @carbon/react v1.84), and the issue has been fixed.
Based on the interactive variant (since -default variant does not have any active status) SR announces the status of the steps appropriately with the current step in inactive state.
